src/main/java/server/application/controllers/Controller.java

```diff
@@ -4,5 +4,11 @@
 import server.core.response.ResponseBuilder;
 
 public interface Controller {
-    void execute(Request request, ResponseBuilder response);
+    void handleGet(Request request, ResponseBuilder response);
+    void handlePost(Request request, ResponseBuilder response);
+    void handlePut(Request request, ResponseBuilder response);
+    void handlePatch(Request request, ResponseBuilder response);
+    void handleDelete(Request request, ResponseBuilder response);
+    void handleOptions(Request request, ResponseBuilder response);
+    void handleHead(Request request, ResponseBuilder response);
```

Comment:
maybe drop "handle" from these function names? I know, I know function names should be verbs, but it's kind of redundant.